py-polars/polars/expr/list.py

```diff
@@ -459,7 +459,7 @@ def contains(
         item = parse_as_expression(item, str_as_lit=True)
         return wrap_expr(self._pyexpr.list_contains(item))
 
-    def join(self, separator: str) -> Expr:
+    def join(self, separator: str | Expr) -> Expr:
```

Comment:
This should be `IntoExpr` and then in the function we can call `parse_as_expr` (IIRC).

Reply:
Perf!